libnavigation-base/src/test/java/com/mapbox/navigation/base/route/RouteExclusionsTest.kt

```diff
@@ -35,6 +35,7 @@ class RouteExclusionsTest {
                     add(
                         mockk {
                             every { routeId } returns "$it"
+                            every { routerOrigin } returns com.mapbox.navigator.RouterOrigin.ONBOARD
```

Comment:
I think you wouldn't need to get through many tests and fix mock setup for DTO if we used factories like [here](https://github.com/mapbox/mapbox-navigation-android/pull/5767/files#diff-f9cd0342c0c73c28b4e1998bdb1be641e387915f52a9e873f0089cdff15bd962R322)

Reply:
That's true, but the problem with this specific setup is that the factory would need to be based on `libnavigation-base`. So I can't add it to `libtesting-utils` because it would create a circular dependency.

The only way I could see is to create a new test utils modules that is only based on `libnavigation-base`, and didn't want to pull the trigger at the time, refs https://github.com/mapbox/mapbox-navigation-android/pull/5587#discussion_r832403393.